### DISCUSSION TITLE
Lesson 11: metmask  showing broken minted nft on anvil chain

### DISCUSSION THREAD

**Discussion by deep0072** [Cyfrin/foundry-full-course-cu#536] [Q&A]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/46425800/d58fa03f-7b34-44f2-b84c-ecb7f01b98c0)

**Comment by shikhar229169** [comment]
Ensure that the token URI returned is correct. Even if there is a single mistake in the uri, it will not show up as the json file get's messed up.

**Reply by deep0072** [reply]
How to ensure? Do I need to console log it into test file?

**Reply by shikhar229169** [reply]
Yes do console logs in your test file and you can view them by adding -vv at the end of forge test command.
After that copy the token URI and search it in browser.
If it shows json, then copy the image link from there and search it up in browser. If both works fine then it is ok.
But if there is error with any one then yeah you need to check that up.

**Reply by deep0072** [reply]
I have checked there is no error. When check token uri and image uri in browser. Even i have cleared nonce from metamask.

**Comment by theirrationalone** [comment]
First, clear all the recently deployed contracts... (that out and broadcast folders). 

Second, open metamask delete all broken NFTs. 

Third, into metamask open settings -> advanced settings -> clear all the tab data and nonce data. 

Fourth, come back to vs code, start anvil.

Fifth, check metamask is connected with anvil Or not? If not please connect it(follow Patrick to connect) 

Fifth, open another terminal and Deploy NFT again (Make sure you're deploying NFT after testing it, after cross checking uri) with broadcast. 

Sixth, Add deployed nft, into metamask. 

Done! 


Should show NFT correctly now.. 

It's actually  metamask's anvil connection glitch, Or we can say it a bug, Can we?? Idk.

**Reply by shikhar229169** [reply]
Yes it can be that what you mentioned.
Sometimes metamask malfunctions. 
But in my case I messed up with the token URI in the contract and it showed the same error - unnamed collection.

**Reply by theirrationalone** [reply]
> Yes it can be that what you mentioned.
> Sometimes metamask malfunctions. 
> But in my case I messed up with the token URI in the contract and it showed the same error - unnamed collection.

Yep i mentioned this problem too, in this underlying comment tree 🌲

**Comment by theirrationalone** [comment]
> First, clear all the recently deployed contracts... (that out and broadcast folders). 
> 
> Second, open metamask delete all broken NFTs. 
> 
> Third, into metamask open settings -> advanced settings -> clear all the tab data and nonce data. 
> 
> Fourth, come back to vs code, start anvil.
> 
> Fifth, check metamask is connected with anvil Or not? If not please connect it(follow Patrick to connect) 
> 
> Fifth, open another terminal and Deploy NFT again (Make sure you're deploying NFT after testing it, after cross checking uri) with broadcast. 
> 
> Sixth, Add deployed nft, into metamask. 
> 
> Done! 
> 
> 
> Should show NFT correctly now.. 
> 
> It's actually  metamask's anvil connection glitch, Or we can say it a bug, Can we?? Idk. 
> 
> 
> 

Or it could be a naming problem, check your broadcast folder -> latest deployment -> inspect name Or title (what mentioned in it idk) into transaction details, and organise if it is unnamed Or untitled?? If it's then add your NFT contract name Or title there.

**Reply by deep0072** [reply]
What kind of name you are talking about. Btw I have followed all steps. Still showing same unnamed

![IMG20230805003739.jpg](https://github.com/Cyfrin/foundry-full-course-f23/assets/46425800/53f02116-142e-4cb5-8379-5cd3884ee79e)

**Reply by theirrationalone** [reply]
This **HIGHLIGHTED** One. also follow left menu bar to trace directory.
![Screenshot (83)](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/d62b3401-99d4-461e-8ead-113471463413)

**Reply by theirrationalone** [reply]
If that's okay too in your project, And you have put & cross checked the **URI**, cleared the none, then It's definitely anvil and its connection to metamask.... I survived from this, I took 2-3 try(s) to get that NFT visible....,


# Try to deploy the same NFT twice and then grab the address of Second one's (Latest NFT) after that, link that latest address to Metamask.

**Reply by deep0072** [reply]
Contact name also there. Yeah I need to check again.

**Reply by theirrationalone** [reply]
> Contact name also there. Yeah I need to check again. 

So if you're sure about name or title then... 

Deploy same nft two time and use latest one to import it into metamask.

**Reply by deep0072** [reply]
still same issue

**Reply by theirrationalone** [reply]
> still same issue

Try to reset everything and then deploy it 3x/4x/5x times and and import latest always into metamask.

You can check the issue into anvil's terminal after each deployment.... Check there if anvil is able to recognize nft Or contract Name... 

I deployed 2-3x times after that i got my nft into metamask. Really annoying....

You can move on to next chapter though, and leave the discussion open until you get the answer.

**Reply by deep0072** [reply]
done everything but still same issue. i am moving on btw.

**Reply by shikhar229169** [reply]
Try with these NFTs:
https://github.com/shikhar229169/NFT-Marketplace-Foundry/tree/master/images/svgs

**Comment by cebidhem** [comment]
Hi @deep0072 ,

I was struggling as well on this, and after few hours I noticed that I forgot a separator `,` in the json we are building in `function tokenURI()`.

Made the metadata right and now it's working as expected!
